File: backend/internal/routes/finansije.go
package routes

import (
	"beleg-app/backend/internal/handlers"

	"github.com/gin-gonic/gin"
)

// RegisterFinanceRoutes registruje sve rute za finansije na datu grupu.
// Grupa mora biti već zaštićena AuthMiddleware-om (npr. protected /api).
// Provera uloge (admin/blagajnik) je unutar handlera.
func RegisterFinanceRoutes(g *gin.RouterGroup) {
	g.GET("/finansije/transakcije/:id", handlers.GetTransakcijaByID)
	g.DELETE("/finansije/transakcije/:id", handlers.DeleteTransakcija)
	g.GET("/finansije", handlers.GetTransakcije)
	g.GET("/finansije/dashboard", handlers.GetDashboard)
	g.POST("/finansije", handlers.CreateTransakcija)
	g.GET("/finansije/clanarine", handlers.GetClanarine)
	g.POST("/finansije/clanarina", handlers.PostClanarinaPlati)
}

File: src/services/finansije.ts
import api from './api'

export async function fetchFinansijeDashboard(params: URLSearchParams) {
  const res = await api.get(`/api/finansije/dashboard?${params}`)
  return res.data
}

export async function fetchClanarine(godina: number) {
  const res = await api.get(`/api/finansije/clanarine?godina=${godina}`)
  return res.data
}

export async function createTransakcija(payload: Record<string, unknown>) {
  await api.post('/api/finansije', payload)
}

export async function createClanarina(payload: Record<string, unknown>) {
  await api.post('/api/finansije/clanarina', payload)
}

export async function deleteTransakcija(id: number) {
  await api.delete(`/api/finansije/transakcije/${id}`)
}

export async function fetchTransakcijaById<T = unknown>(id: number) {
  const res = await api.get<T>(`/api/finansije/transakcije/${id}`)
  return res.data
}
